python/test/unit/language/test_core.py

```diff
@@ -5457,3 +5457,39 @@ def kernel(input):
 
     kernel[grid](input)
     assert torch.all(input == torch.tensor(grid, device=device))
+
+
+# -----------------------
+# test extern functions
+# -----------------------
+
+
+@pytest.mark.parametrize("dtype_str", ['float32', 'float64'])
+def test_math_extern(dtype_str):
```

Comment:
I assume this test should be marked as nvidia-only?

Reply:
Indeed, added
```
    if not is_cuda():
        pytest.skip('math_extern test only works on CUDA')
```

Reply:
what part is Nvidia specific? libdevice is now there for AMD as well

Reply:
Glad to know it, modified it accordingly, and enabled the test on both hip and cuda.

Reply:
Ah, sorry, I thought that "libdevice" meant it was nvidia-specific.